src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -66,7 +66,6 @@ class FilterButton extends React.Component<Props, State> {
     onClose = () => {
         this.setState({
             isMenuOpen: false,
-            transientConditions: [],
```

Comment:
This change seems to conflict with this line -> "const wasFlyoutClosed = transientConditions.length === 0;"

What does this fix accomplish?

Reply:
Just added another change.

transientConditions should not be set to an empty array in the ```onClose``` method, it should instead be set to an empty array in the ``` applyFilters``` method. This is due to us needing to hydrate transientConditions only when filters have been applied. Otherwise, just by closing the flyout does not mean we should hydrate transientConditions.